sdk/appcenter-push/src/test/java/com/microsoft/appcenter/push/PushTest.java

```diff
@@ -761,6 +763,45 @@ public void verifyEnqueueCalledOnNewAuthToken() {
         verify(channel, times(2)).enqueue(any(Log.class), anyString(), anyInt());
     }
 
+    @Test
+    public void verifyEnqueueCalledOnNewUserId() {
+        String mockUserId1 = "userId1";
+        String mockUserId2 = "userId2";
+        Push push = Push.getInstance();
+        Channel channel = mock(Channel.class);
+        doNothing().when(channel).enqueue(any(Log.class), anyString(), anyInt());
+        start(push, channel);
+        push.onTokenRefresh("push-token");
+        ArgumentCaptor<PushInstallationLog> log = ArgumentCaptor.forClass(PushInstallationLog.class);
+        UserIdContext.getInstance().setUserId(mockUserId1);
+        verify(channel, times(2)).enqueue(log.capture(), anyString(), anyInt());
```

Comment:
we need to put the first verify at the right place otherwise we don't know why we have the second call.